c61665245.lua

```diff
@@ -10,117 +10,72 @@ function c61665245.initial_effect(c)
+	return c:IsRace(rac) and c:IsCanBeSpecialSummoned(e,0,tp,false,false)
+end
+function c61665245.sptg(e,tp,eg,ep,ev,re,r,rp,chk)
 	local zone=e:GetHandler():GetLinkedZone(1-tp)
 	if chk==0 then return Duel.GetLocationCount(1-tp,LOCATION_MZONE,tp,LOCATION_REASON_TOFIELD,zone)>0
 		and Duel.IsExistingMatchingCard(c61665245.spfilter1,tp,LOCATION_HAND,0,1,nil,e,tp,zone) end
-	Duel.SetOperationInfo(0,CATEGORY_SPECIAL_SUMMON,nil,1,tp,LOCATION_HAND)
+	Duel.SetOperationInfo(0,CATEGORY_SPECIAL_SUMMON,nil,2,tp,LOCATION_HAND+LOCATION_DECK)
 end
-function c61665245.spop1(e,tp,eg,ep,ev,re,r,rp)
+function c61665245.spop(e,tp,eg,ep,ev,re,r,rp)
+	local c=e:GetHandler()
 	local zone=e:GetHandler():GetLinkedZone(1-tp)
 	if Duel.GetLocationCount(1-tp,LOCATION_MZONE,tp,LOCATION_REASON_TOFIELD,zone)<=0 then return end
 	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_SPSUMMON)
-	local g=Duel.SelectMatchingCard(tp,c61665245.spfilter1,tp,LOCATION_HAND,0,1,1,nil,e,tp,zone)
-	if g:GetCount()>0 then
-		Duel.SpecialSummon(g,0,tp,1-tp,false,false,POS_FACEUP_DEFENSE,zone)
-	end
-end
-function c61665245.spfilter2(c,e,tp,lg)
-	return c:IsFaceup() and lg:IsContains(c) and Duel.IsExistingMatchingCard(c61665245.spfilter3,tp,LOCATION_DECK,0,1,nil,e,tp,c:GetRace())
-end
-function c61665245.spfilter_chkc(c,e,tp,lg,rc)
-	return c:IsFaceup() and lg:IsContains(c) and (c:GetRace()&rc)==rc
-end
-function c61665245.spfilter3(c,e,tp,rac)
-	if not c:IsRace(rac) then return false end
-	local ok=false
-	for p=0,1 do
-		local zone=e:GetHandler():GetLinkedZone(p)&0xff
-		ok=ok or (Duel.GetLocationCount(p,LOCATION_MZONE,tp,LOCATION_REASON_TOFIELD,zone)>0 and c:IsCanBeSpecialSummoned(e,0,tp,false,false,POS_FACEUP_DEFENSE,p,zone))
-	end
-	return ok
-end
-function c61665245.sptg2(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
-	local c=e:GetHandler()
-	local zone={}
-	zone[0]=c:GetLinkedZone(0)
-	zone[1]=c:GetLinkedZone(1)
-	local lg=c:GetLinkedGroup()
-	if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) and c61665245.spfilter_chkc(chkc,e,tp,lg,e:GetLabel()) end
-	if chk==0 then return Duel.IsExistingTarget(c61665245.spfilter2,tp,LOCATION_MZONE,LOCATION_MZONE,1,nil,e,tp,lg) end
-	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_TARGET)
-	local g=Duel.SelectTarget(tp,c61665245.spfilter2,tp,LOCATION_MZONE,LOCATION_MZONE,1,1,nil,e,tp,lg)
-	Duel.SetOperationInfo(0,CATEGORY_SPECIAL_SUMMON,nil,1,tp,LOCATION_DECK)
-	e:SetLabel(g:GetFirst():GetRace())
-end
-function c61665245.spop2(e,tp,eg,ep,ev,re,r,rp)
-	local c=e:GetHandler()
-	local tc=Duel.GetFirstTarget()
-	if not c:IsRelateToEffect(e) or not tc:IsRelateToEffect(e) or tc:IsFacedown() then return end
-	local zone={}
-	local flag={}
-	for p=0,1 do
-		zone[p]=c:GetLinkedZone(p)&0xff
-		local _,flag_tmp=Duel.GetLocationCount(p,LOCATION_MZONE,tp,LOCATION_REASON_TOFIELD,zone[p])
-		flag[p]=(~flag_tmp)&0x7f
+	local tc=Duel.SelectMatchingCard(tp,c61665245.spfilter1,tp,LOCATION_HAND,0,1,1,nil,e,tp,zone):GetFirst()
+	local function makeDisable(dc)
```

Comment:
should be separate

Reply:
it's should not separate, because the function referenced `c` which is this card itself.